keps/sig-node/3066-subsecond-probes/README.md

```diff
@@ -0,0 +1,859 @@
+
+We expect no non-infra related flakes in the last month as a GA graduation criteria.
+-->
+
+- <test>: <link to test coverage> TBD
+
+- Existing node-e2e test: `k8s.io/kubernetes/test/e2e/common/container_probe.go`
+
+Enhanced with additional test cases, see the buckets added in [the draft implementation](https://github.com/kubernetes/kubernetes/pull/107958/commits)
+
+
+
+### Graduation Criteria
+
+#### Alpha
+
+- Feature implemented behind a feature flag
+- Initial e2e tests completed and enabled
+
+#### Beta
+
+- Gather feedback from developers and surveys regarding success and scalability
+- Complete a performance analysis/review and adjust
+- Additional tests are in Testgrid and linked in KEP
+
+#### GA
+
+- Collect examples of real-world usage
+- More rigorous forms of testing—e.g., downgrade tests and scalability tests
+- Allowing time for feedback
+
+### Upgrade / Downgrade Strategy
+
+Design will be tested to be hardened against upgrade/downgrade issues.
+
+### Version Skew Strategy
+
+Design consideration has been made for and will be tested to ensure forward and backward version skew compatibility.
+
+## Production Readiness Review Questionnaire
+
+<!--
+
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable; can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/1194-prod-readiness.
+
+The production readiness review questionnaire must be completed and approved
+for the KEP to move to `implementable` status and be included in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
+### Feature Enablement and Rollback
+
+_This section must be completed when targeting alpha to a release._
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [ X ] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name: SubSecondProbes
+    - Components depending on the feature gate:
+		kube-apiserver and kubelet
+
+* **Does enabling the feature change any default behavior?**
+
+  No
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+  Yes, see the [drop test](https://github.com/psschwei/kubernetes/blob/1ca40771e26b96d6121c49b19c2cbe6466694c7a/pkg/api/pod/util_test.go#L1029)
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+  No breaking changes
```

Comment:
If you previously set the new fields, and due to problems rolled it back, does the field gets cleaned or not? What happens after you enable this back, in that case?

Reply:
ok will add details..